### DISCUSSION TITLE
K value in Range search

### DISCUSSION THREAD

**Discussion by xiaokj37** [milvus-io/milvus#28676] [Q&A and General discussion]
In range search, is the “limit” parameter mandatory? Can I get all the records by just setting “radius” and “range_filter”？

**Comment by MrPresent-Han** [comment]
No, there's a built-in limit to restrict that at most 16384 vectors can be retrieved at one time, which is a protection for memory. If you want to get all similar vectors in the range of [radius, range_filter], I advised you to use pymilvus search_iterator, which has been supported since version 2.3.0.

**Reply by xiaokj37** [reply]
Thanks for your replay. The point I am trying to make is whether "limit" is a required value in range search.

**Comment by MrPresent-Han** [comment]
in the latest developing version of pymilvus, SearchIterator can implement what you want, you can install pymilvus with following command:
pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pymilvus==2.3.3.post1.dev2
